src/lib.rs

```diff
@@ -77,14 +77,14 @@ fn macho_get_section<'a>(macho: &mach::MachO<'a>, section_name: &str) -> Option<
     let segment_name = "__DWARF";
     let section_name = macho_translate_section_name(section_name);
 
-    for segment in &*macho.segments {
+    for segment in &macho.segments {
         if let Ok(name) = segment.name() {
             if name == segment_name {
-                if let Ok(sections) = segment.sections() {
-                    for section in sections {
+                for section in segment {
```

Comment:
I'm not keen on the use of `IntoIterator` here, it seems a bit too magic. A `Segment` isn't just a collection of `Sections`; there's some other info in the segment header too. I would prefer `segment.sections()` to return a `SectionIterator` instead.

Reply:
It's mostly a collection of sections ;)

So sections is still there, tho it will allocate instead of using the iterator (like last time): https://docs.rs/goblin/0.0.11/goblin/mach/segment/struct.Segment.html#method.sections

I added cause it just looks cleaner to me, and it's usually what I do with segments. Anyway feel free to re-add. I can change to a Result<Box<Iterator>> next release to
Have basically the same semantics (all errors collected into single one) but without the extra allocations of you want ?

Reply:
The current behavior is perfect, I just expected it to be the behavior of `sections` too, and I didn't see the `IntoIterator` when I first wrote this code, ie I saw the `fn sections` and assumed that was the only thing available for getting the sections from the segment.